src/core/qgismobileapp.cpp

```diff
@@ -917,6 +927,56 @@ void QgisMobileapp::print( int layoutIndex )
   PlatformUtilities::instance()->open( destination );
 }
 
+void QgisMobileapp::printAtlasFeatures( const QString &layoutName, const QList<long long> featureIds )
+{
+  const QList<QgsPrintLayout *> printLayouts = mProject->layoutManager()->printLayouts();
+  QgsPrintLayout *layoutToPrint = nullptr;
+  for( QgsPrintLayout *layout : printLayouts )
+  {
+    if ( layout->name() == layoutName )
+    {
+      layoutToPrint = layout;
+      break;
+    }
+  }
+
+  if ( !layoutToPrint || !layoutToPrint->atlas() )
+    return;
+
+  QStringList ids;
+  for( const auto id : featureIds )
+  {
+    ids << QString::number( id );
+  }
+
+  QString error;
+  layoutToPrint->atlas()->setFilterExpression( QStringLiteral( "$id IN (%1)" ).arg( ids.join( ',' ) ), error );
```

Comment:
Would it be possible to keep the existing filter expression in the layout and add the feature filter with an "AND"?

Reply:
@signedav , I thought about it but I don't think it makes sense here. We _only_ want to print the selected feature(s) we're requesting.

Reply:
We also risk ending up with zero features. Say the filter has a generic expression like "MY_FIELD" != 'VALUE_1' and the user is requesting to print an atlas feature that is excluded by the pre-existing expression, that'd be confusing and  ultimately not what the user wants.

Reply:
Fair enough.

Another thing: While testing it did not work that it only printed the selected features. Like here (careful, it's not the same like in the clickup ticket) 
[GCK-Rto.zip](https://github.com/opengisch/QField/files/6062264/GCK-Rto.zip)

Reply:
@signedav , can you attach a screencast or steps to reproduce?

Reply:
see #1732 - to double check I created a clean atlas (might be used later in the demo projects) - you can find there a screencast and reproduce it with these projects (advanced beefarming has single file and simple beefarming has one file per feature).